UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -450,50 +477,99 @@ private void RefreshAll()
 		public void RollRandomCharacter()
 		{
 			// Randomise gender
-			var changeGender = (UnityEngine.Random.Range(0, 2) == 0);
-			if (changeGender)
+			Type gender = typeof(BodyType);
+			Array genders = gender.GetEnumValues();
+			int index = UnityEngine.Random.Range(0,3);
+			currentCharacter.BodyType = (BodyType)genders.GetValue(index);
+
+			//Randomises player name and age.
+			switch (currentCharacter.BodyType)
 			{
-				//OnGenderChange();
+				case BodyType.Male:
+					currentCharacter.Name = StringManager.GetRandomMaleName();
+					break;
+				case BodyType.Female:
+					currentCharacter.Name = StringManager.GetRandomFemaleName();
+					break;
+				default:
+					currentCharacter.Name = StringManager.GetRandomName(Gender.NonBinary);  //probably should get gender neutral names?
+					break;																	//for now it will pick from both the male and female name pools
 			}
-
-			// Select a random value from each dropdown
-			// hairDropdown.value = UnityEngine.Random.Range(0, hairDropdown.options.Count - 1);
-			// facialHairDropdown.value = UnityEngine.Random.Range(0, facialHairDropdown.options.Count - 1);
-			// underwearDropdown.value = UnityEngine.Random.Range(0, underwearDropdown.options.Count - 1);
-			// socksDropdown.value = UnityEngine.Random.Range(0, socksDropdown.options.Count - 1);
-				switch (currentCharacter.BodyType)
-				{
-					case BodyType.Male:
-						currentCharacter.Name = StringManager.GetRandomMaleName();
-						break;
-					case BodyType.Female:
-						currentCharacter.Name = StringManager.GetRandomFemaleName();
-						break;
-					default:
-						currentCharacter.Name = StringManager.GetRandomName(Gender.NonBinary);  //probably should get gender neutral names? 
-						break;																	//for now it will pick from both the male and female name pools
-				}
-			// Randomise rest of data
-			// currentCharacter.EyeColor = "#" + ColorUtility.ToHtmlStringRGB(UnityEngine.Random.ColorHSV());
-			// currentCharacter.HairColor = "#" + ColorUtility.ToHtmlStringRGB(UnityEngine.Random.ColorHSV());
-			// currentCharacter.SkinTone = availableSkinColors[UnityEngine.Random.Range(0, availableSkinColors.Count - 1)];
 			currentCharacter.Age = UnityEngine.Random.Range(19, 78);
 
-			//RefreshAll();
+			//Randomises character skin tones from a select list of colors.
+			currentCharacter.SkinTone = "#" + ColorUtility.ToHtmlStringRGB(availableSkinColors[UnityEngine.Random.Range(0, availableSkinColors.Count - 1)]);
```

Comment:
for this check if there is any available colours if there is just randomly generate a colour make sure alphas is always one,

This is so if you don't specify skin tones the player can choose any, and if you do specify skin tones the player has to choose one of those

Reply:
if you paid attention to my commits and code you'd know that not only are we not generating new random colors (seriously, read the comment RIGHT ABOVE this one line of code) but alpha issues are no longer a thing because they use your race tones lists now that you did not set and I've made sure their alpha values are always at one.

Reply:
"This is so if you don't specify skin tones the player can choose any, and if you do specify skin tones the player has to choose one of those"

Reply:
if the list is empty that means the player is free to choose any skin tone they want, any RGB value,  however if it's populated then they are restricted to those one specified in the list

Reply:
if you don't specify any skin tones in the race tones list, the entire thing breaks up and shits itself. (Like how it was already broken initially when you merged without setting skin tones for humans)
if you specify skin tones, it works just fine and you get the right skin tones for your character's race.

I don't get what you're trying to push here, do you want the game to generate red humans and ignore your own feature that you coded in yourself? I'm too sick and don't have the energy to argue at all, no changes are going to be done until we fix character spawning first as that's more important issue than this.

Reply:
the character spawning is fixed, 
"(Like how it was already broken initially when you merged without setting skin tones for humans)"
was it? I know the randomised button was broken but never got around to fixing it, 
the point is if you have a species for example you don't care about the colour as the species coder, for example lizards, they can have any colour for scales, so this just allows an easy way of selecting option of let the user pick,
for the human ones I never got around to setting up the skin colours so was just left blank

Reply:
> the character spawning is fixed,
No, it isn't. Pull the PR that you're supposed to be REVIEWING and check for yourself. I've even said it up before that characters aren't spawning correctly with their customizations and you ignored that.

> was it? I know the randomised button was broken but never got around to fixing it,
You didn't test your own change that you pushed that partially broke character customization and you shouldn't push a change that you didn't even know works or not. You shouldn't also comment out code that isn't working with your changes then say "oh welp, i'll fix it some other day" then push a massive list of changes like this because guess what? I'm the one now who's fixing it and have problems working with your un-understandable undocumented code that makes no sense with how poorly undefined your variable names are and your dynamic code hell. I'm all for making everything dynamic through code but with the way your approaching things and your undocumented messy code it's fucking hard to work on anything and fix issues.

> the point is if you have a species for example you don't care about the colour as the species coder, for example lizards, they can have any colour for scales, so this just allows an easy way of selecting option of let the user pick,
for the human ones I never got around to setting up the skin colours so was just left blank

You could have just said "oh make sure that other races have access to all colors instead of a specific list of colors with full random generation" but even your code review requests are like your god damn code it self, it's hard to fucking understand.

Bod, for the love of god. Stay away from my health and sprite related PRs for the next few days unless you're offering a fix/solution to a problem or are going to document or explain your changes because right now I'm working in the dark and you aren't helping at all.

Sorry if I sound aggressive, no hard or bad feelings between us. But your terrible code is giving me anger issues whenever I think about it.

Reply:
Hey, tone it down. What is this unneeded hostility? If you’re feeling sick then go away from the computer and come back when you’re feeling better. The reviewers are not your enemies, stop antagonizing on GitHub

Reply:
but do you know what I'm getting at
currentCharacter.SkinTone randomise 
availableSkinColors = empty list = allow any colour
availableSkinColors = populated list = choose from list

Reply:
I know, randomise completely got messed up by my work

Reply:
I'm not antagonizing or being hostile against Bod, Like I said, nothing bad between us. I'm just being honest about what I think about this whole situation.

Reply:
yeah, my documentation is terrible that's true, I would have done it by now but need to hurry with finishing final year project

Reply:
> but do you know what I'm getting at

Yeah I get you now, will add it to my to do list and will work on soon when I get back on unity.

> yeah, my documentation is terrible that's true, I would have done it by now but need to hurry with finishing final year project

good luck with your project and make us proud, our little pog champ.